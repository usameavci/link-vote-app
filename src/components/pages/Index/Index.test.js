import React from "react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import PageIndex from "./Index";
import LinkContextProvider from "../../../contexts/LinkContext";

describe("Index page component tests", () => {
  it("renders without crashing", async () => {
    const component = (
      <Router>
        <LinkContextProvider>
          <PageIndex />
        </LinkContextProvider>
      </Router>
    );

    let getByTestId;

    await act(async () => ({ getByTestId } = await render(component)));

    expect(getByTestId("page-index")).toBeInTheDocument();
  });

  it("matches snapshots", async () => {
    const component = (
      <Router>
        <LinkContextProvider>
          <PageIndex />
        </LinkContextProvider>
      </Router>
    );

    let tree;

    await renderer.act(async () => (tree = renderer.create(component)));

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
