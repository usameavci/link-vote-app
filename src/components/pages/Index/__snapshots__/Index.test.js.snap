// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page component tests matches snapshots 1`] = `
<div
  className="page-index"
  data-testid="page-index"
>
  <div
    className="page-index-block"
  >
    <a
      className="link-submit"
      data-testid="link-submit"
      href="/submit"
      onClick={[Function]}
    >
      <div
        className="link-submit-icon"
      >
        <i
          className="fa fa-plus"
        />
      </div>
      <div
        className="link-submit-title"
      >
        SUBMİT A LİNK
      </div>
    </a>
  </div>
  <div
    className="page-index-block"
  >
    <div
      className="link-sort"
      data-testid="link-sort"
    >
      <div
        className="common-select"
        data-testid="common-select"
      >
        <div
          className="common-select-label"
          onClick={[Function]}
        >
          <span
            className="common-select-label-inner"
          >
            Creation Date
          </span>
          <span
            className="common-select-label-icon"
          >
            <i
              className="fa fa-caret-down"
            />
          </span>
        </div>
        <div
          className="common-select-options"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="common-select-option"
            onClick={[Function]}
          >
            Creation Date
          </div>
          <div
            className="common-select-option"
            onClick={[Function]}
          >
            Most Voted (Z -&gt; A)
          </div>
          <div
            className="common-select-option"
            onClick={[Function]}
          >
            Less Voted (A -&gt; Z)
          </div>
        </div>
      </div>
    </div>
    <div
      className="link-list"
      data-testid="link-list"
    >
      <div
        data-testid="link-list-empty"
      >
        <div
          className="text-center"
        >
          There is a no links here
        </div>
        <div
          className="text-center"
        >
          <button
            onClick={[Function]}
          >
            Load Example Data
          </button>
        </div>
      </div>
    </div>
    <div
      className="link-pagination"
      data-testid="link-pagination"
    >
      
    </div>
  </div>
</div>
`;
